routes/user.js

```diff
@@ -0,0 +1,165 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const Error = require("../models/Error");
+const Joi = require("@hapi/joi");
+const jwtService = require("../services/JwtService");
+
+const userService = require("../services/UserService");
+
+router
+    .route("/")
+    .post((req, res) => {
+        let {
+            zprn,
+            firstName,
+            lastName,
+            password,
+            department,
+            role,
+        } = req.body;
+        const { error } = validateData(
+            zprn,
+            firstName,
+            lastName,
+            password,
+            department,
+            role
+        );
+        if (error != undefined) {
+            res.status(400).send(new Error("BAD_REQUEST", error.message));
+        } else {
+            bcrypt
+                .genSalt(10)
+                .then((salt) => {
+                    bcrypt
+                        .hash(password, salt)
+                        .then((hash) => {
+                            userService
+                                .addUser(
+                                    zprn,
+                                    firstName,
+                                    lastName,
+                                    hash,
+                                    department,
+                                    role
+                                )
+                                .then(() => {
+                                    res.status(201).json({
+                                        code: "CREATED",
+                                        msg: "User added!",
+                                    });
+                                })
+                                .catch((e) => {
+                                    res.status(500).json({
+                                        code: "INTERNAL_SERVER_ERROR",
+                                        msg: e,
+                                    });
+                                });
+                        })
+                        .catch((er) => {
+                            res.status(500).json({
+                                code: "INTERNAL_SERVER_ERROR",
+                                msg: er,
+                            });
+                        });
+                })
+                .catch((err) => {
+                    res.status(500).json({
+                        code: "INTERNAL_SERVER_ERROR",
+                        msg: err,
+                    });
+                });
+        }
+    })
+    .patch((req, res) => {
```

Comment:
Also shouldn't we have like a parameter to update a user in the `url`? Accepting it in body seems different. 
Check out [this](https://zulipchat.com/api/update-user) link for how we can also do it. `api/user/{user_id}` is the url that will be more suitable?